lib/src/main/kotlin/dev/chrisbanes/snapper/LazyList.kt

```diff
@@ -54,15 +93,17 @@ import kotlin.math.roundToInt
  * @param maximumFlingDistance Block which returns the maximum fling distance in pixels.
  * The returned value should be > 0.
  */
-@ExperimentalSnapperApi
 @Composable
+@Deprecated("maximumFlingDistance has been replaced with snapIndex")
+@Suppress("DEPRECATION")
+@ExperimentalSnapperApi
 public fun rememberSnapperFlingBehavior(
     lazyListState: LazyListState,
     snapOffsetForItem: (layoutInfo: SnapperLayoutInfo, item: SnapperLayoutItemInfo) -> Int = SnapOffsets.Center,
     endContentPadding: Dp = 0.dp,
     decayAnimationSpec: DecayAnimationSpec<Float> = rememberSplineBasedDecay(),
     springAnimationSpec: AnimationSpec<Float> = SnapperFlingBehaviorDefaults.SpringAnimationSpec,
-    maximumFlingDistance: (SnapperLayoutInfo) -> Float = SnapperFlingBehaviorDefaults.MaximumFlingDistance,
+    maximumFlingDistance: (SnapperLayoutInfo) -> Float,
```

Comment:
it is actually a binary breaking deprecation. I would recommend keeping the default value so the signature of the function is not changing. compiler should already be smart enough and use the correct overload when maximumFlingDistance wasn't provided as the non deprecated overloads have a higher weight at the moment of resolution. or you removed it because it didn't work well for you?

Reply:
The compiler can't disambiguate from my testing, even with Kotlin 1.6.10. Need to think on this more

Reply:
maybe it doesn't work for the trailing lambdas? if we define this new param as not the last param will it work?

Reply:
Just managed to get it to work with a manual overload for the `snapIndex` param (rather than using a default value). PTAL